DotnetFoundation/DotnetFoundation.Api/Controllers/AuthenticationController.cs

```diff
@@ -27,11 +29,25 @@ public AuthenticationController(IAuthenticationService authenticationService)
     public async Task<ActionResult<BaseResponse<AuthenticationResponse>>> RegisterAsync(RegisterRequest request)
     {
         BaseResponse<AuthenticationResponse> response = new(ResponseStatus.Fail);
+        try
+        {
+            response.Data = await _authenticationService.RegisterAsync(request).ConfigureAwait(false);
+            response.Status = ResponseStatus.Success;
 
-        response.Data = await _authenticationService.RegisterAsync(request).ConfigureAwait(false);
-        response.Status = ResponseStatus.Success;
-
-        return Ok(response);
+            return Ok(response);
+        }
+        catch (IdentityUserException ex)
+        {
+            response.Message = ex.Message;
```

Comment:
there should be a method called `ModelState.AddModelError()` which can be used to get the messages in the errors array. Suggestion to use it.

Reply:
We need to install Mvc package in the services , as per clean architecture we shouldn't have this dependency of package in services

Reply:
used actionContextAccessor to get Model State reference